tcms/bugs/tests/test_api.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init
+
+from xmlrpc.client import ProtocolError
+
+from tcms.bugs.models import Bug
+from tcms.bugs.tests.factory import BugFactory
+from tcms.rpc.tests.utils import APITestCase
+
+
+class TestRemove(APITestCase):
+    """Test Bug.remove"""
+
+    def setUp(self):
+        super(TestRemove, self).setUp()
+
+        self.bug0 = BugFactory(status=False)
+        self.bug1 = BugFactory(status=True)
+        self.bug2 = BugFactory(status=True)
+
+    def test_remove_bugs(self):
+        self.rpc_client.Bug.remove({'status': 'True'})
+        bugs = Bug.objects.all()
+        self.assertIn(self.bug0, bugs)
+        self.assertNotIn(self.bug1, bugs)
+        self.assertNotIn(self.bug2, bugs)
```

Comment:
This scenario is correct but the api client (self.api_user) has all of the possible permissions in the system and that should also be very granular. 

You can leave that as a functional test and then duplicate together with the negative scenario to test permissions. 

- For positive scenario user needs only 1 permission - the 1 which is checked by the decorator
- For negative scenario user needs all permissions from DB except the 1 checked by the decorator

^^^ there have been copy & paste errors in the past. That's why we need this granularity. Here you have found a similar one in the documentation.